### PR TITLE
Dev/mip 62/mip engine integration with the mip portalbackend

### PR DESCRIPTION
**Pull request by KFilippopolitis**

### CODE REVIEW

File: src/main/java/eu/hbp/mip/models/DTOs/MIPEngineAlgorithmDTO.java [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+    @SerializedName("crossvalidation")
+    private String crossvalidation;
+
+    @SerializedName("inputdata")
+    private InputdataDTO inputdata;
+
+    @Getter
+    @Setter
+    public static class ParameterDTO {
```
Comment (ThanKarab) [inline src/main/java/eu/hbp/mip/models/DTOs/MIPEngineAlgorithmDTO.java:40]: nit: Rename to MIPEngineAlgorithmParameterDTO?

File: src/main/java/eu/hbp/mip/models/DTOs/MIPEngineAlgorithmDTO.java [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+        @SerializedName("max")
+        private String max;
+
+        @SerializedName("default_value")
+        private String default_value;
+
+        @SerializedName("enums")
+        private List<String> enums;
+
+        public ExaremeAlgorithmRequestParamDTO convertToexaremeAlgorithmRequestDTO(String name){
```
Comment (ThanKarab) [inline src/main/java/eu/hbp/mip/models/DTOs/MIPEngineAlgorithmDTO.java:69]: After a discussion we had, we decided to remove all "converters" and change them to constructors.

File: src/main/java/eu/hbp/mip/models/DTOs/MIPEngineAlgorithmDTO.java [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+            exaremeAlgorithmRequestParamDTO.setValueMultiple(this.multiple);
+            exaremeAlgorithmRequestParamDTO.setValueMin(this.min);
+            exaremeAlgorithmRequestParamDTO.setValueMax(this.max);
+            return exaremeAlgorithmRequestParamDTO;
+        }
+    }
+
+    @Getter
+    @Setter
+    public static class InputdataDTO {
```
Comment (ThanKarab) [inline src/main/java/eu/hbp/mip/models/DTOs/MIPEngineAlgorithmDTO.java:88]: nit: MIPEngineInputDataDTO?

File: src/main/java/eu/hbp/mip/models/DTOs/MIPEngineAlgorithmDTO.java [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+            exaremeAlgorithmRequestParamDTOS.add(this.pathology.convertToExaremeAlgorithmRequestDTO("pathology"));
+            exaremeAlgorithmRequestParamDTOS.add(this.datasets.convertToExaremeAlgorithmRequestDTO("dataset"));
+            exaremeAlgorithmRequestParamDTOS.add(this.filter.convertToExaremeAlgorithmRequestDTO("filter"));
+            return exaremeAlgorithmRequestParamDTOS;
+        }
+    }
+
+    @Getter
+    @Setter
+    public static class InputDataDetailDTO {
```
Comment (ThanKarab) [inline src/main/java/eu/hbp/mip/models/DTOs/MIPEngineAlgorithmDTO.java:117]: nit: prefix with MIPEngine?

File: src/main/java/eu/hbp/mip/models/DTOs/MIPEngineTabularVisualizationDTO.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+import lombok.Getter;
+import lombok.Setter;
+
+import java.util.HashMap;
+import java.util.List;
+
+@Getter
+@Setter
+@AllArgsConstructor
+public class MIPEngineTabularVisualizationDTO {
```
Comment (ThanKarab) [inline src/main/java/eu/hbp/mip/models/DTOs/MIPEngineTabularVisualizationDTO.java:13]: This does not belong to MIPEngine but to portal-backend / frontend. I think no prefix is needed for that... Not sure though.

File: src/main/resources/application.yml [hunk 6]
```diff
@@ -34,6 +34,8 @@ spring:
 
 ### EXTERNAL SERVICES ###
 services:
+  mipengine:
+    algorithmsUrl: "http://192.168.124.129:5000/algorithms"
```
Comment (ThanKarab) [inline src/main/resources/application.yml:38]: nit: localhost?

**Review by ThanKarab — CHANGES_REQUESTED** [review]
Nice work!

Quick summary:
1) Remove all converters and create constructors.
2) Move Galaxy calls to a separate "GalaxyService".
3) Rename all internal classes with the proper prefix (MIPEngine or Exareme)

minor: use localhost instead of internal ip.

File: docker/README.md [hunk 7]
```diff
@@ -24,6 +24,8 @@ To use this image, you need a running instance of PostgreSQL and to configure th
 
 
 ### EXTERNAL SERVICES ###
+* MIPENGINE_URL: URL to MIPENGINE server. Default is "http://192.168.124.129:5000" .
```
[the remark this answers is not in the document]
Comment (KFilippopolitis) [reply docker/README.md:27]: We need mip engine to be dockerized to access the localhost.

**Review by ThanKarab — APPROVED** [review]

**Pull request merged**